pkg/api/types.go

```diff
@@ -1175,6 +1180,11 @@ func (m *MasterProfile) IsUbuntu() bool {
 	return m.IsUbuntu1604() || m.IsUbuntu1804()
 }
 
+// IsUbuntuNonVHD returns true if the distro uses a base Ubuntu image
+func (m *MasterProfile) IsUbuntuNonVHD() bool {
+	return m.IsUbuntu() && !m.IsVHDDistro()
```

Comment:
this is equivalent to the already existing `IsUbuntu` function, I don't think we need a new one.

Reply:
```
	switch m.Distro {
	case AKS, Ubuntu, ACC1604:
		return true
```

Reply:
```
// IsUbuntu returns true if the master profile distro is any ubuntu distro
func (m *MasterProfile) IsUbuntu() bool {
	return m.IsUbuntu1604() || m.IsUbuntu1804()
}
```

Reply:
We are pasting the same thing. :)

`IsUbuntu()` returns true if the distro is `AKS` or `AKS1804`. That doesn't fulfill the requirements of "ubuntu non-VHD".

Reply:
Oh wow I totally misread that, my bad

Reply:
I thought `IsUbuntu1604` was literally the `Ubuntu1604` distro... that's a bit confusing.

Reply:
I still think we want this to be `return m.Distro == Ubuntu || m.Distro == Ubuntu1804` to exclude ACC1604 and ONLY include the vanilla Ubuntu image explicitly.

Reply:
Why do we want to exclude `ACC1604`?

Reply:
I don't know if we do but it's a special case https://github.com/Azure/aks-engine/blob/master/docs/topics/sgx.md and a custom image that the Azure Confidential Compute team maintains. @dmitsh might be able to tell us if we should apply CIS hardening for those.

Reply:
I think it should be out of scope for now and we can optionally opt it in later if it turns out they do want it. We don't have good testing around it though so we risk breaking it with this PR if we don't validate that scenario.

Reply:
As long as it's an Ubuntu 16.04-LTS or 18.04-LTS base let's include it, it should work fine. These changes should not effect any k8s app layer behaviors, they should simply provide a better hardened OS underpinning for Ubuntu based on CIS recommendations.

Reply:
Making it out of scope essentially means bringing it out of the `IsUbuntu` conveniences. We can't reasonably manually work around the `ACC1604` case any other way.

Reply:
I see what you're saying. We also need `IsUbuntuNonVHD` to be equivalent to 
```
$FULL_INSTALL_REQUIRED; then
  if [[ $OS == $UBUNTU_OS_NAME ]]; then
```
otherwise the file permission step will fail.